tests/ssl-opt.sh

```diff
@@ -695,6 +695,17 @@ run_test() {
     CLI_EXPECT="$3"
     shift 3
 
+    # Expected exit code for the server in response to SIGTERM
+    # Expect 1 for gnutls-serv, otherwise default to expecting 0
```

Comment:
And openssl (at least version 1.0.2g, which is the one we use on most of the CI) exits with error code 143 (that's from the signal number).

Since neither openssl nor gnutls exit with status 0, and we usually assume that they're working correctly, I think we should ignore their exit status. There's a risk that if we pass an unsupported option, they'll default to doing something other than what we expected, but the exist status doesn't help with that.

And so we should only check the exit status of mbedtls, which is something we do care about, for example if it had a memory leak which is reported by exiting with a nonzero status.